suite-native/module-send/src/components/CoinControl/SwitchCoinControlButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { AccountKey } from '@suite-common/wallet-types';
+import { Button, HStack, Text } from '@suite-native/atoms';
+import { Icon } from '@suite-native/icons';
+import { Translation } from '@suite-native/intl';
+import {
+    RootStackParamList,
+    SendStackParamList,
+    SendStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+import { useUtxoSelection } from '../../hooks/useUtxoSelection';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SendStackParamList,
+    SendStackRoutes.SendUtxo,
+    RootStackParamList
+>;
+
+type SwitchCoinControlButtonProps = {
+    accountKey: AccountKey;
+    amount?: string;
+};
+
+export const SwitchCoinControlButton = ({ accountKey, amount }: SwitchCoinControlButtonProps) => {
+    const { selectedUtxos, totalSelectedAmount, isCoinControlEnabled } = useUtxoSelection();
+    const navigation = useNavigation<NavigationProps>();
+
+    const isMissingUtxos = isCoinControlEnabled && amount && totalSelectedAmount.isLessThan(amount);
+
+    const openCoinControlScreen = () => {
+        navigation.navigate(SendStackRoutes.SendUtxo, {
+            accountKey,
+            amount,
+        });
+    };
+
+    const colorScheme = useMemo(() => {
+        if (isCoinControlEnabled) {
+            return isMissingUtxos ? 'yellowBold' : 'primary';
+        }
+
+        return 'tertiaryElevation0';
+    }, [isCoinControlEnabled, isMissingUtxos]);
+
+    const textColor = isCoinControlEnabled && !isMissingUtxos ? 'textOnPrimary' : 'textOnTertiary';
+
+    return (
+        <Button
+            testID="switch-coin-control-button"
+            onPress={openCoinControlScreen}
+            colorScheme={colorScheme}
+        >
+            <HStack alignItems="center" spacing="sp8">
+                <Icon size="mediumLarge" color={textColor} name="coins" />
+                <Text color={textColor}>
+                    <Translation id="moduleSend.coinControl.cta" />
+                    {isCoinControlEnabled ? ` • ${selectedUtxos.length}` : ''}
+                </Text>
+            </HStack>
+        </Button>
```

Comment:
You can use `viewLeft` prop for the icon. Then the HStack and Icon components can be removed.
```suggestion
        <Button
            testID="switch-coin-control-button"
            onPress={openCoinControlScreen}
            colorScheme={colorScheme}
            viewLeft="coins"
        >
            <Text color={textColor}>
                <Translation id="moduleSend.coinControl.cta" />
                {isCoinControlEnabled ? ` • ${selectedUtxos.length}` : ''}
            </Text>
        </Button>
```

Reply:
yes! I did not notice that Button has this prop.

Reply:
2cebc088c7c43abe87ef6cbad186c1e1db5c6daa